xdsl/transforms/stencil_shape_minimize.py

```diff
@@ -50,7 +76,14 @@ class StencilShapeMinimize(ModulePass):
 
     name = "stencil-shape-minimize"
 
+    restrict: tuple[int, ...] | None
```

Comment:
```suggestion
    restrict: tuple[int, ...] | None = None
```
This would set a default value to not restrict the size.